tests/unit/test_client.py

```diff
@@ -111,6 +112,16 @@ def test_client_signature_override(self, request_signer):
         request_signer.assert_called_with(
             mock.ANY, mock.ANY, mock.ANY, 'foo', mock.ANY, mock.ANY)
 
+    @mock.patch('botocore.client.RequestSigner')
```

Comment:
This might not be possible, but can we add a test that's more "direct"?  I see that this test is essentially checking that we forward the value of config.signature_version to the request signer, but it'd be great if we had something that verified, if you set the signature version to UNSIGNED, then we do not sign the request.  I get there's another level of indirection here with the RequestSigner and the auth classes, but it seems possible to regress on this feature without any unit tests failing because we're never actually testing anything that explicitly sets the value to `botocore.UNSIGNED`.